scm/driver/stash/git.go

```diff
@@ -77,37 +74,28 @@ func (s *gitService) ListCommits(ctx context.Context, repo string, opts scm.Comm
 
 func (s *gitService) ListTags(ctx context.Context, repo string, opts scm.ListOptions) ([]*scm.Reference, *scm.Response, error) {
 	namespace, name := scm.Split(repo)
-	path := fmt.Sprintf("rest/api/1.0/projects/%s/repos/%s/tags?%s", namespace, name, encodeListOptions(opts))
+	endpoint := fmt.Sprintf("rest/api/1.0/projects/%s/repos/%s/tags?%s", namespace, name, encodeListOptions(opts))
 	out := new(branches)
-	res, err := s.client.do(ctx, "GET", path, nil, &out)
-	if !out.pagination.LastPage.Bool {
-		res.Page.First = 1
-		res.Page.Next = opts.Page + 1
-	}
+	res, err := s.client.do(ctx, "GET", endpoint, nil, out)
+	copyPagination(out.pagination, res)
```

Comment:
can we move this change (and others in this file) to a separate pull request?  This will keep this PR isolated only to the List functionality.